### ISSUE TITLE
update to firefox 85.0.1

### ISSUE THREAD

**Issue by git-bruh** [kiss-community/repo#9]
new patch: https://github.com/kiss-community/repo-main/blob/da5c3312abd8c92818be3b07c37a73a4478e58a3/extra/firefox/patches/no-gtk2.patch

build fails with
```
13:33.14 xpcom/build/Services.cpp.stub
13:33.60 xpcom/components
13:35.00 /tmp/32170/build/firefox/toolkit/xre/nsEmbedFunctions.cpp: In function 'nsresult XRE_InitChildProcess(int, char**, const XREChildData*)':
13:35.00 /tmp/32170/build/firefox/toolkit/xre/nsEmbedFunctions.cpp:530:3: error: 'XInitThreads' was not declared in this scope
13:35.00   530 |   XInitThreads();
13:35.00       |   ^~~~~~~~~~~~
13:35.23 xpcom/ds
13:35.93 make[4]: *** [/tmp/32170/build/firefox/config/rules.mk:676: nsEmbedFunctions.o] Error 1
13:35.93 make[3]: *** [/tmp/32170/build/firefox/config/recurse.mk:72: toolkit/xre/target-objects] Error 2
13:35.93 make[3]: *** Waiting for unfinished jobs....
13:51.67    Compiling webrender v0.61.0 (/tmp/32170/build/firefox/gfx/wr/webrender)
14:00.01    Compiling webrender_bindings v0.1.0 (/tmp/32170/build/firefox/gfx/webrender_bindings)
16:11.69    Compiling gkrust v0.1.0 (/tmp/32170/build/firefox/toolkit/library/rust)
19:58.48     Finished release [optimized] target(s) in 19m 39s
19:58.98 make[2]: *** [/tmp/32170/build/firefox/config/recurse.mk:34: compile] Error 2
19:58.98 make[1]: *** [/tmp/32170/build/firefox/config/rules.mk:355: default] Error 2
19:58.98 make: *** [client.mk:89: build] Error 2
19:58.98 0 compiler warnings present.
```

`X11/Xlib.h` and `-lX11` related

**Comment by subsonik** [comment]
I've been trying to get 85.0.1 working under KISS, also. I can give this new patch a try a little later.

Regarding the error, I guess X11/Xlib.h needs including in nsEmbedFunctions.cpp?

I'm not a C/C++ developer, so I've been struggling with this. :(

**Comment by git-bruh** [comment]
@subsonik I tried doing that aswell but then the build breaks with lots of other errors. and Xlib.h is probably included already in one of the included files. We dont touch any of those

**Comment by subsonik** [comment]
I've made some progress, but it's just build error after build error, currently this one:

```
 9:08.51 warning: In file included from glsl-optimizer/src/compiler/glsl/ast_array_index.cpp:24:
 9:08.51 warning: In file included from glsl-optimizer/src/compiler/glsl/ast.h:29:
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/glsl_parser_extras.h:34:
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/glsl_symbol_table.h:28:
 9:08.52 warning: In file included from /home/john/src/firefox-86.0/obj-x86_64-pc-linux-musl/dist/stl_wrappers/new:52:
 9:08.52 warning: /home/john/src/firefox-86.0/obj-x86_64-pc-linux-musl/dist/system_wrappers/new:3:15: fatal error: 'new' file not found
 9:08.52 warning: #include_next <new>
 9:08.52 warning:               ^~~~~
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl_types.cpp:26:
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/glsl_parser_extras.h:34:
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/glsl_symbol_table.h:28:
 9:08.52 warning: In file included from /home/john/src/firefox-86.0/obj-x86_64-pc-linux-musl/dist/stl_wrappers/new:52:
 9:08.52 warning: /home/john/src/firefox-86.0/obj-x86_64-pc-linux-musl/dist/system_wrappers/new:3:15: fatal error: 'new' file not found
 9:08.52 warning: #include_next <new>
 9:08.52 warning:               ^~~~~
 9:08.52 warning: 1 error generated.
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/ast_expr.cpp:24:
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/ast.h:29:
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/glsl_parser_extras.h:34:
 9:08.52 warning: In file included from glsl-optimizer/src/compiler/glsl/glsl_symbol_table.h:28:
 9:08.52 warning: In file included from /home/john/src/firefox-86.0/obj-x86_64-pc-linux-musl/dist/stl_wrappers/new:52:
 9:08.52 warning: /home/john/src/firefox-86.0/obj-x86_64-pc-linux-musl/dist/system_wrappers/new:3:15: fatal error: 'new' file not found
 9:08.52 warning: #include_next <new>
 9:08.52 warning:               ^~~~~
 9:08.52 warning: 1 error generated.
 9:08.52 warning: 1 error generated.
 9:08.60 dom/media/bridge
 9:09.28 error: build failed
 9:09.29 make[4]: *** [/home/john/src/firefox-86.0/config/makefiles/rust.mk:327: force-cargo-library-build] Error 101
 9:09.29 make[3]: *** [/home/john/src/firefox-86.0/config/recurse.mk:72: toolkit/library/rust/target] Error 2
```

This one, and a similar "#include_next <type_traits>" error.

**Comment by git-bruh** [comment]
It was due to this code block

```
-#if defined(XP_UNIX)
+#if defined(MOZ_X11)
 #  include <stdio.h>
-#  if defined(MOZ_X11)
+#  if 0
 #    include <X11/Xlib.h>
 #    include <X11/Xutil.h>
 #    include "X11UndefineNone.h"
```

this was the first thing i had noticed, but i didn't bother changing it because it had been same in the earlier patches... (It prevents `X11/Xlib.h` from being included)
